README.md

```diff
@@ -1,69 +1,273 @@
+3. The operator also reconciles every 5 minutes against all Ingresses on the cluster.
 
-```sh
-kubectl apply -f config/samples/
-```
+### Azure permissions and RBAC
+Permissions needed for the operator to run properly are as follows.
 
-2. Build and push your image to the location specified by `IMG`:
-	
-```sh
-make docker-build docker-push IMG=<some-registry>/reply-urls-operator:tag
-```
-	
-3. Deploy the controller to the cluster with the image specified by `IMG`:
+#### Azure permissions
+The Operator needs to be able to read and write to the App Registrations and can be added via the `API Permissions` tab on the App Registration itself.
 
-```sh
-make deploy IMG=<some-registry>/reply-urls-operator:tag
+* API Permissions: `Application.ReadWrite.All` (Type: Application)
+
+**Note:** If you are running the cluster locally you can use the az cli to authenticate with Azure as long as your user is able to Read and Write to the App Registration that the Operator is configured to manage.
+
+#### Cluster RBAC
+
+The Operator needs the permissions below to work properly.
+
+| resources     | verbs            |
+|---------------|------------------|
+| replyurlsyncs | get, list, watch |
+| ingresses     | get, list, watch |
+
+All the RBAC files can be found in the `config/rbac` folder. They are created using markers in the Operators Go code, markers for RBAC can be found in `controllers/ingress_controller.go` and look similar to below.
+
+```go
+//+kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch
 ```
 
-### Uninstall CRDs
+[More information on RBAC markers](https://book.kubebuilder.io/reference/markers/rbac.html) 
+
+## Running the Operator
+
+Before deploying anything you will need an Azure App Registration that will have its Reply URLs updated by the Operator and either the same App Registration or a separate one that has the right permissions to update the App Registration's Reply URLs.
+
+[Instructions on creating an Azure App Registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
+
+You will need to take note of the Object ID of the App Registration that will be managed by the Operator and the Client/Application ID, Client Secret and Tenant ID of the App Registration that will be used to Authenticate.
+
+Once you have created the app registration you will need to give it the correct permissions. This can be done by clicking on the API permissions tab in the Azure portal whilst viewing the app registration. You then need to click on Add a permission and then Microsoft Graph, click on Application permissions and search for Application. Add the Application.ReadWrite.All permission and click Add permissions, then click the Grant admin consent button.
```

Comment:
sentence per line makes reading non rendered files easier and reviewing changes simpler.

```suggestion
Once you have created the app registration you will need to give it the correct permissions.
This can be done by clicking on the API permissions tab in the Azure portal whilst viewing the app registration.
You then need to click on Add a permission and then Microsoft Graph, click on Application permissions and search for Application.
Add the `Application.ReadWrite.All` permission and click Add permissions, then click the Grant admin consent button.
```